src/backend/core/api/viewsets.py

```diff
@@ -1102,7 +1102,7 @@ def versions_list(self, request, *args, **kwargs):
     @drf.decorators.action(
         detail=True,
         methods=["get", "delete"],
-        url_path="versions/(?P<version_id>[0-9a-z-]+)",
+        url_path=r"versions/(?P<version_id>[^/]{1,1024})",
```

Comment:
```suggestion
        url_path=r"versions/(?P<version_id>[A-Za-z0-9._+\-=~]{1,1024})",
```

I would suggest something less permissive. Your proposal afraid me a bit and I don't want to put a regex so permissive in production. What I suggest is less permissive and I think enough to match your requirement.

WDYT ?